### ISSUE TITLE
'reset all' in .muttrc will cause multiple options to have the wrong default values

### ISSUE THREAD

**Issue by aradici** [neomutt/neomutt#654]
The root cause is the position of set_default/restore_default, which was moved in 40f81d1 (thanks flatcap for finding it out!)

The issue was reported in Debian in
http://bugs.debian.org/865822
http://bugs.debian.org/865842

A patch is available here:
https://anonscm.debian.org/cgit/pkg-mutt/mutt.git/plain/debian/patches/upstream/865822-restore-defaults.patch?id=e71b6806d29343bc6c95b609ce742fcd23daaf62

I'm not sure if this impacted surrounding code so it should probably be checked; I'm issuing this for an emergency release but I'm happy to change it with a better one, as long as it doesn't break alias, tmpdir, etc

**Comment by flatcap** [comment]
I've pushed the patch into master as quite a few variables are affected.
I might make a new release for this.

For (my own) reference, `mutt_init()` sets some global variables, then performs a loop through **all** of them.

In `init.h`, the "default" values for string variables didn't seems to be set consistently.
(I was considering tidying them).  Some were `UL ""` others were `0` (zero).  There's a difference.

```c
/* init.c mutt_init() */
/* Set standard defaults */
for (i = 0; MuttVars[i].option; i++)
{ 
  set_default(&MuttVars[i]);
  restore_default(&MuttVars[i]);
}
```

If the variable in init.h has a initial value of:
- `0` then `set_default()` will copy from the associated global variable into the "initial" value
- `""` then `set_default()` will not do anything

`restore_default()` copies the "initial" value into the global variable -- initialising **all** global variables.

/cc @sileht

**Comment by flatcap** [comment]
Fixed by 1cf8ca6

**Comment by aradici** [comment]
FYI pushed to debian git HEAD
https://anonscm.debian.org/cgit/pkg-mutt/mutt.git/commit/?id=46caca2d121ab77455ab2e34cbc24e054e16fe7f
